src/extension.ts

```diff
@@ -164,3 +177,31 @@ function convertLogLevel(commonLogLevel: string) {
             return "FINE";
     }
 }
+
+function resolvePotentialProjectFiles() {
+    let editors = vscode.window.visibleTextEditors.filter((e) => e.document.languageId === "java");
+    const activeEditor = vscode.window.activeTextEditor;
+    if (activeEditor.document.languageId === "java") {
+        const index = editors.indexOf(activeEditor);
+        editors = editors.splice(index, 1);
+        editors = [activeEditor].concat(editors);
+    }
+    if (editors) {
+        const projects = editors.map((e) => {
+            const workspaceRoot = vscode.workspace.rootPath;
+            const fullpath = e.document.fileName;
```

Comment:
when the editor opens a virtual document, the fileName info may be null. should test this case.

Reply:
nice catch!